cmd/common/progress.go

```diff
@@ -0,0 +1,78 @@
+package common
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+)
+
+// ProgressOption configures the behaviour of ProgressWriter.
+type ProgressOption func(*ProgressWriter)
+
+// WithMessage sets the status prefix shown before percentage/KiB.
+func WithMessage(msg string) ProgressOption {
+	return func(p *ProgressWriter) { p.msg = msg }
+}
+
+// WithFrequency sets how often updates are printed.
+func WithFrequency(freq time.Duration) ProgressOption {
+	return func(p *ProgressWriter) { p.freq = freq }
+}
+
+// WithOutput sets the destination writer (defaults to os.Stderr).
+func WithOutput(w io.Writer) ProgressOption {
+	return func(p *ProgressWriter) { p.out = w }
+}
+
+// ProgressWriter is an io.Writer that prints a simple progress bar.
+// It is safe to use with io.TeeReader:
+//
+//	pw := common.NewProgressWriter(resp.ContentLength, common.WithMessage("Downloading..."))
+//	io.Copy(dst, io.TeeReader(src, pw))
```

Comment:
A single space should work fine according to https://go.dev/doc/comment.

```suggestion
//  pw := common.NewProgressWriter(resp.ContentLength, common.WithMessage("Downloading..."))
//  io.Copy(dst, io.TeeReader(src, pw))
```